Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -12,137 +13,175 @@ struct NestingRule: Rule {
         nonTriggeringExamples: NestingRuleExamples.nonTriggeringExamples,
         triggeringExamples: NestingRuleExamples.triggeringExamples
     )
+}
+
+private struct Levels {
+    private var typeLevel: Int = -1
+    private var functionLevel: Int = -1
+    private var functionOrNotStack = Stack<Bool>()
 
-    private let omittedStructureKinds = SwiftDeclarationKind.variableKinds
-        .union([.enumcase, .enumelement])
-        .map(SwiftStructureKind.declaration)
+    private var lastIsFunction: Bool { functionOrNotStack.peek() == true }
 
-    private struct ValidationArgs {
-        var typeLevel: Int = -1
-        var functionLevel: Int = -1
-        var previousKind: SwiftStructureKind?
-        var violations: [StyleViolation] = []
+    mutating func validate(
```

Comment:
Oh sorry, I missed the fact that access to the internal levels is needed in this method. In that case, `private(set)` was actually just fine. The `validate` method doesn't really fit into the `Levels` type otherwise as its sole purpose is to keep track of levels - nothing more.

Reply:
Thanks! I moved `validate` logic into Visitor's method.